streams/src/main/java/org/apache/kafka/streams/kstream/JoinWindows.java

```diff
@@ -65,17 +66,34 @@
  */
 public final class JoinWindows extends Windows<Window> {
 
+    private static final long DEFAULT_MAINTAIN_DURATION_MS = 24 * 60 * 60 * 1000L; // default: one day
+    private final long maintainDurationMs;
```

Comment:
We can't remove the retention time implementation in the superclass, but we don't actually have to use it, either. Since `maintainDurationMs` in the superclass is private, we can simply override the setter and getter and instead implement it as an immutable field.